test/e2e/federated-service.go

```diff
@@ -106,58 +117,60 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 			}
 			framework.Logf("%d clusters are Ready", len(contexts))
 
-			clusterClientSets = make([]*release_1_3.Clientset, len(clusterList.Items))
-			for i, cluster := range clusterList.Items {
-				framework.Logf("Creating a clientset for the cluster %s", cluster.Name)
+			// clusters = make([]cluster, len(clusterList.Items))
+			clusters = map[string]*cluster{}
+			primaryClusterName = clusterList.Items[0].Name
```

Comment:
Also, the ordering of clusterlist depends on the ordering of the list returned by the API call to list clusters, which is not, as far as I know, well defined.  I think that we need a more definitive means of determining the "primary cluster".  It's the cluster in which the federation control plane is running (irrespective of where it appears in the list).